File: bluetoothk/src/main/java/com/mozhimen/bluetoothk/helpers/BluetoothKStateChangeReceiver.java
package com.mozhimen.bluetoothk.helpers;

import android.bluetooth.BluetoothDevice;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.mozhimen.bluetoothk.BluetoothK;

/**
 * Created by lee on 5/30/16.
 */
public class BluetoothKStateChangeReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();

        Log.i("BluetoothStateChange", "action = " + action);

        if (action.equals(BluetoothDevice.ACTION_ACL_DISCONNECTED)) {
            BluetoothDevice device = (BluetoothDevice) intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
            String mac = device.getAddress();
            Log.i("BluetoothStateChange", "mac = " + mac);
            BluetoothK.executeBluetoothDisconnectedCallback(mac);
        }
    }
}

File: bluetoothk/src/main/java/com/mozhimen/bluetoothk/helpers/BluetoothKStateChangeReceiver.kt
package com.mozhimen.bluetoothk.helpers

import android.bluetooth.BluetoothDevice
import android.content.Context
import android.content.Intent
import android.os.Parcelable
import com.mozhimen.kotlin.elemk.android.content.bases.BaseBroadcastReceiver
import com.mozhimen.kotlin.utilk.android.util.d
import com.mozhimen.kotlin.utilk.android.util.w
import com.mozhimen.bluetoothk.BluetoothK
import com.mozhimen.kotlin.elemk.android.bluetooth.cons.CBluetoothDevice

/**
 * @ClassName BluetoothKStateChangeReceiver
 * @Description TODO
 * @Author Mozhimen & Kolin Zhao
 * @Date 2023/8/17 11:03
 * @Version 1.0
 */
class BluetoothKStateChangeReceiver : BaseBroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        "onReceive action = $action".d(TAG)

        if (action == CBluetoothDevice.ACTION_ACL_DISCONNECTED) {
            val device = intent.getParcelableExtra<Parcelable>(CBluetoothDevice.EXTRA_DEVICE) as? BluetoothDevice?
            if (device == null) {
                "onReceive device == null".w(TAG)
                return
            }
            val mac = device.address
            "onReceive mac = $mac".d(TAG)
            BluetoothK.instance.executeBluetoothDisconnectedCallback(mac)
        }
    }
}